libraries/AP_Notify/Buzzer.h

```diff
@@ -38,19 +36,16 @@ class Buzzer: public NotifyDevice
     /// on - turns the buzzer on or off
     void on(bool on_off);
 
-    // Patterns - how many beeps will be played
-    enum BuzzerPattern {
-        NONE = 0,
-        SINGLE_BUZZ = 1,
-        DOUBLE_BUZZ = 2,
-        GPS_GLITCH = 3, // not used
-        ARMING_BUZZ = 4,
-        BARO_GLITCH = 5,
-        EKF_BAD = 6
-    };
+    // Patterns - how many beeps will be played; read from
+    // left-to-right, each bit represents 10ms
+    static const uint32_t    SINGLE_BUZZ = 0b10000000000000000000000000000000UL;
+    static const uint32_t    DOUBLE_BUZZ = 0b10100000000000000000000000000000UL;
+    static const uint32_t    ARMING_BUZZ = 0b11111111111111111111111111111100UL; // 3s
+    static const uint32_t    BARO_BUZZ   = 0b10101010000000000000000000000000UL;
+    static const uint32_t      EKF_BAD   = 0b11110111011010000000000000000000UL;
 
     /// play_pattern - plays the defined buzzer pattern
-    void play_pattern(BuzzerPattern pattern_id);
+    void play_pattern(const uint32_t pattern);
```

Comment:
Is there any reason for this and `on` methods and the patterns to be public? I only found it being used from the Buzzer class.

Reply:
Moved to private

Reply:
`on` and patterns are still public.

Reply:
Fixed.